### PR TITLE
tests: Add a WRITE_WATCH test

### PR DESCRIPTION
**Pull request by Joshua-Ashton**

### CODE REVIEW

File: tests/d3d12.c [hunk 1]
```diff
@@ -48996,6 +48996,118 @@ static void test_cbv_hoisting_dxil(void)
...
+    desc.no_render_target = true;
+    desc.no_pipeline = true;
+    desc.no_root_signature = true;
+    if (!init_test_context(&context, &desc))
+        return;
+
+    memset(&heap_properties, 0, sizeof(heap_properties));
+
+    resource_desc.Dimension = D3D12_RESOURCE_DIMENSION_BUFFER;
+    resource_desc.Alignment = 0;
```
Comment (HansKristian-Work) [inline tests/d3d12.c:49029]: Is this okay?

File: tests/d3d12.c [hunk 2]
```diff
@@ -48996,6 +48996,118 @@ static void test_cbv_hoisting_dxil(void)
...
+    resource_desc.DepthOrArraySize = 1;
+    resource_desc.MipLevels = 1;
+    resource_desc.Format = DXGI_FORMAT_UNKNOWN;
+    resource_desc.SampleDesc.Count = 1;
+    resource_desc.SampleDesc.Quality = 0;
+    resource_desc.Layout = D3D12_TEXTURE_LAYOUT_ROW_MAJOR;
+
+    buffer = NULL;
+    heap_properties.Type = D3D12_HEAP_TYPE_DEFAULT;
+    resource_desc.Flags = D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS;
```
Comment (HansKristian-Work) [inline tests/d3d12.c:49041]: This is not allowed on UPLOAD heap.

File: tests/d3d12.c [hunk 3]
```diff
@@ -48996,6 +48996,118 @@ static void test_cbv_hoisting_dxil(void)
...
+    ok(dirty_addresses[2] == (void*)&map_ptr[5 * page_size], "Expected dirty address 2 to be %p, got %p\n",
+            (void*)&map_ptr[5 * page_size], dirty_addresses[2]);
+    ok(dirty_addresses[3] == (void*)&map_ptr[9 * page_size], "Expected dirty address 3 to be %p, got %p\n",
+            (void*)&map_ptr[9 * page_size], dirty_addresses[3]);
+
+done:
+    free(dirty_addresses);
+
+    if (buffer)
+        ID3D12Resource_Release(buffer);
```
Comment (HansKristian-Work) [inline tests/d3d12.c:49107]: Forgot to release context.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Nits. Did you run this against D3D12 validator?

**Comment by Joshua-Ashton** [comment]
D3D12 Validation? No. 

I ran it on native D3D12 and VKD3D-Proton and it all passed however.

File: tests/d3d12.c (hunk 1 above)
Comment (Joshua-Ashton) [reply tests/d3d12.c:49029]: Alignment = 0? Yeah.

`Alignment must be 64KB (D3D12_DEFAULT_RESOURCE_PLACEMENT_ALIGNMENT) or 0, which is effectively 64KB.`

File: tests/d3d12.c (hunk 3 above)
Comment (Joshua-Ashton) [reply tests/d3d12.c:49107]: Fixed.

**Review by HansKristian-Work — APPROVED** [review]
LGTM

**Pull request merged**